addons/website/static/tests/builder/website_builder/menu_data.test.js

```diff
@@ -195,4 +197,103 @@ describe("EditMenuDialog", () => {
         expect(".oe_menu_editor").toHaveCount(1);
         expect(".js_menu_label").toHaveText("Top Menu Item");
     });
+
+    test("clicking save in the EditMenuDialog should not clear the editor changes", async () => {
+        const { getEditor } = await setupWebsiteBuilder(
+            // Using tel: as link to avoid having to mock fetching metadata for link preview
+            // This does not influence the test in any way
+            `<ul class="top_menu">
+                <li>
+                    <a class="nav-link" href="tel: 123" contenteditable="true">
+                        <span>Top Menu Item</span>
+                    </a>
+                </li>
+            </ul>
+            <section>
+                <p>Text</p>
+            </section>`,
+            {
+                openEditor: true,
+            }
+        );
+
+        const fakeORMService = {
+            call: (model, method, args) => {
+                if (method === "save" && model === "ir.ui.view") {
+                    expect(args[1]).toInclude("Text, more text");
+                    hasSaved = true;
```

Comment:
I prefer to use expect.step("save") and not hasSaved.
hasSaved don t check if you have multi save